PostHog/PostHogSDK.swift

```diff
@@ -524,6 +526,52 @@ let maxRetryDelay = 30.0
         remoteConfig?.setPersonPropertiesForFlags(allProperties)
     }
 
+    /// Returns fresh default device and app properties for feature flag evaluation.
+    /// This ensures feature flags can use current properties like $app_version, $os_version, etc.
+    /// These properties are computed fresh each time they're needed.
+    private func getDefaultPersonProperties() -> [String: Any] {
+        guard config.setDefaultPersonProperties else { return [:] }
+        guard isEnabled() else { return [:] }
+
+        let staticContext = context?.staticContext() ?? [:]
+
+        var defaultProperties: [String: Any] = [:]
+
+        // App information
+        if let appVersion = staticContext["$app_version"] {
+            defaultProperties["$app_version"] = appVersion
+        }
+        if let appBuild = staticContext["$app_build"] {
+            defaultProperties["$app_build"] = appBuild
+        }
+
+        // Operating system information
+        if let osName = staticContext["$os_name"] {
+            defaultProperties["$os_name"] = osName
+        }
+        if let osVersion = staticContext["$os_version"] {
+            defaultProperties["$os_version"] = osVersion
+        }
+
+        // Device information
+        if let deviceType = staticContext["$device_type"] {
+            defaultProperties["$device_type"] = deviceType
+        }
+
+        // Localization - read directly to avoid expensive dynamicContext call
+        if #available(iOS 16.0, macOS 13.0, tvOS 16.0, watchOS 9.0, *) {
```

Comment:
nit: extract this to a [utils](https://github.com/PostHog/posthog-ios/tree/main/PostHog/Utils) method and reuse here and [here](https://github.com/PostHog/posthog-ios/blob/03803c80ae67695667a8106c98af92caac94e81f/PostHog/PostHogContext.swift#L181-L189)